v0/destinations/mautic/utils.js

```diff
@@ -0,0 +1,298 @@
+    (context.traits && context.traits?.subscriptionStatus)
+  ) {
+    const status =
+      traits && traits?.subscriptionStatus
+        ? message.traits?.subscriptionStatus
+        : message.context.traits?.subscriptionStatus;
+    if (!ALLOWED_SUBSCRIPTION_STATUS.includes(status)) {
+      throw new CustomError("Invalid subscriptionStatus Value.", 400);
+    }
+    set(payload, "subscription_status", status);
+  }
+  if (
+    (traits && traits?.prospectOrCustomer) ||
+    context.traits?.prospectOrCustomer
+  ) {
+    const POC =
+      message.traits?.prospectOrCustomer ||
+      message.context.traits?.prospectOrCustomer;
+    if (!ALLOWED_POC.includes(POC)) {
+      throw new CustomError(
+        "prospectOrCustomer can only be either prospect or customer or null.",
+        400
+      );
+    }
+    set(payload, "prospect_or_customer", POC);
+  }
+  return payload;
+};
+
+//Constructs the address1 and address2 field
+//if address is given as string or object
+const deduceAddressFields = message => {
+  const { traits, context } = message;
+  let address1;
+  let address2;
+  if (
+    (traits !== undefined && traits?.address) ||
+    (context.traits && context.traits?.address)
+  ) {
+    let add =
+      traits && traits?.address
+        ? message.traits.address
+        : message.context.traits.address;
+    if (typeof add === "object") {
+      add = Object.keys(add).reduce(function(res, v) {
+        return res.concat(add[v], " ");
+      }, "");
+    }
+    const validLengthAddress = add.length > 128 ? add.substring(0, 127) : add;
+    address1 = validLengthAddress.substring(0, 63);
+    address2 = validLengthAddress.substring(64, validLengthAddress.length);
+  }
+  return { address1, address2 };
+};
+
+//Validates the generated payload fro specific fields
+const validatePayload = payload => {
+  // checking for message details validations
+  if (payload.email && !validateEmail(payload.email)) {
+    throw new CustomError("Invalid Mail Provided.", 400);
+  }
+  if (payload.phone && !validatePhone(payload.phone)) {
+    throw new CustomError("Invalid Phone No. Provided.", 400);
+  }
+  if (payload.title && !ALLOWED_TITLES.includes(payload.title)) {
+    throw new CustomError("Invalid title provided.", 400);
+  }
+  if (payload.last_active && !isDate(payload.last_active)) {
+    throw new CustomError("Date is Invalid.", 400);
+  }
+  if (payload.state && !(payload.state[0] === payload.state[0].toUpperCase())) {
+    throw new CustomError("State is Invalid.", 400);
+  }
+  return true;
+};
+
+/**
+ *
+ * @param {*} message
+ * @param {*} destination
+ * @returns contactId
+ * If contactId is not provided via externalId, we look for the lookup key
+ * inside webapp config.
+ * We have put two level dynamic mapping here.
+ * If the lookup key is not found we fallback to email. If email is also not provided, we throw error.
+ */
+const searchContactId = async (message, destination, identifyFlag) => {
```

Comment:
We can default the value of `identifyFlag`as true or false and can send the other value explicitly when needed. By doing this we can omit the declaration of the `identifyFlag` variable twice.